diff -Naur old/gdk-pixbuf-2.30.8/configure new/gdk-pixbuf-2.30.8/configure
--- old/gdk-pixbuf-2.30.8/configure	2014-05-27 02:43:18.000000000 +0200
+++ new/gdk-pixbuf-2.30.8/configure	2014-09-23 22:52:22.526714308 +0200
@@ -17666,20 +17666,20 @@
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-   echo "*** The test program compiled, but did not run. This usually means"
-          echo "*** that the run-time linker is not finding GLIB or finding the wrong"
-          echo "*** version of GLIB. If it is not finding GLIB, you'll need to set your"
-          echo "*** LD_LIBRARY_PATH environment variable, or edit /etc/ld.so.conf to point"
-          echo "*** to the installed location  Also, make sure you have run ldconfig if that"
-          echo "*** is required on your system"
-	  echo "***"
-          echo "*** If you have an old version installed, it is best to remove it, although"
-          echo "*** you may also be able to get things to work by modifying LD_LIBRARY_PATH"
-else
-   echo "*** The test program failed to compile or link. See the file config.log for the"
-          echo "*** exact error that occured. This usually means GLIB is incorrectly installed."
-fi
+#if ac_fn_c_try_link "$LINENO"; then :
+#   echo "*** The test program compiled, but did not run. This usually means"
+#          echo "*** that the run-time linker is not finding GLIB or finding the wrong"
+#          echo "*** version of GLIB. If it is not finding GLIB, you'll need to set your"
+#          echo "*** LD_LIBRARY_PATH environment variable, or edit /etc/ld.so.conf to point"
+#          echo "*** to the installed location  Also, make sure you have run ldconfig if that"
+#          echo "*** is required on your system"
+#	  echo "***"
+#          echo "*** If you have an old version installed, it is best to remove it, although"
+#          echo "*** you may also be able to get things to work by modifying LD_LIBRARY_PATH"
+#else
+#   echo "*** The test program failed to compile or link. See the file config.log for the"
+#          echo "*** exact error that occured. This usually means GLIB is incorrectly installed."
+#fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
           CFLAGS="$ac_save_CFLAGS"
@@ -17692,9 +17692,9 @@
      GOBJECT_QUERY=""
      GLIB_MKENUMS=""
      GLIB_COMPILE_RESOURCES=""
-     as_fn_error $? "
-*** GLIB 2.37.6 or better is required. The latest version of
-*** GLIB is always available from ftp://ftp.gtk.org/pub/gtk/." "$LINENO" 5
+#     as_fn_error $? "
+#*** GLIB 2.37.6 or better is required. The latest version of
+#*** GLIB is always available from ftp://ftp.gtk.org/pub/gtk/." "$LINENO" 5
   fi
 
 
diff -Naur old/gdk-pixbuf-2.30.8/gdk-pixbuf/Makefile.in new/gdk-pixbuf-2.30.8/gdk-pixbuf/Makefile.in
--- old/gdk-pixbuf-2.30.8/gdk-pixbuf/Makefile.in	2014-05-27 02:43:20.000000000 +0200
+++ new/gdk-pixbuf-2.30.8/gdk-pixbuf/Makefile.in	2014-09-23 23:46:18.890789213 +0200
@@ -2673,7 +2673,7 @@
 	@true
 
 s-enum-types-h: @REBUILD@ $(gdk_pixbuf_headers) gdk-pixbuf-enum-types.h.template
-	$(AM_V_GEN)( cd $(srcdir) && $(GLIB_MKENUMS) --template gdk-pixbuf-enum-types.h.template \
+	$(AM_V_GEN)(cd $(srcdir) && $(GLIB_MKENUMS) --template gdk-pixbuf-enum-types.h.template \
 		$(gdk_pixbuf_headers) ) > tmp-gdk-pixbuf-enum-types.h \
 	&& (cmp -s tmp-gdk-pixbuf-enum-types.h gdk-pixbuf-enum-types.h || cp tmp-gdk-pixbuf-enum-types.h gdk-pixbuf-enum-types.h ) \
 	&& rm -f tmp-gdk-pixbuf-enum-types.h \
