--- old/grub-0.97/stage2/Makefile.in	2005-05-08 04:42:39.000000000 +0200
+++ new/grub-0.97/stage2/Makefile.in	2014-09-15 23:27:24.446554419 +0200
@@ -407,7 +407,7 @@
 exec_prefix = @exec_prefix@
 host = @host@
 host_alias = @host_alias@
-host_cpu = @host_cpu@
+host_cpu = i386
 host_os = @host_os@
 host_vendor = @host_vendor@
 includedir = @includedir@
@@ -426,6 +426,8 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
+STAGE2_CFLAGS += -fno-reorder-functions
+
 # For test target.
 TESTS = size_test
 noinst_SCRIPTS = $(TESTS)
