diff -Naur old/gtk+-3.14.0/gtk/updateiconcache.c new/gtk+-3.14.0/gtk/updateiconcache.c
--- old/gtk+-3.14.0/gtk/updateiconcache.c	2014-08-15 16:38:05.000000000 +0200
+++ new/gtk+-3.14.0/gtk/updateiconcache.c	2014-10-06 16:44:31.524509760 +0200
@@ -1755,6 +1755,9 @@
   if (!force_update && is_cache_up_to_date (path))
     return 0;
 
+#if GLIB_MAJOR_VERSION * 100 + GLIB_MINOR_VERSION < 236
+  g_type_init ();
+#endif
   replace_backslashes_with_slashes (path);
   build_cache (path);
 
diff -Naur old/gtk+-3.14.0/tests/testgtk.c new/gtk+-3.14.0/tests/testgtk.c
--- old/gtk+-3.14.0/tests/testgtk.c	2014-09-04 20:11:33.000000000 +0200
+++ new/gtk+-3.14.0/tests/testgtk.c	2014-10-06 16:44:31.524509760 +0200
@@ -4966,7 +4966,7 @@
 			"GtkWidget::visible", TRUE,
 			NULL);
 
-#ifdef GDK_WINDOWING_WAYLAND
+#ifdef GDK_WINDOWING_X11
       if (GDK_IS_X11_DISPLAY (gtk_widget_get_display (vbox)))
         cursor_demo = TRUE;
 #endif
diff -Naur old/gtk+-3.14.0/util/Makefile.in new/gtk+-3.14.0/util/Makefile.in
--- old/gtk+-3.14.0/util/Makefile.in	2014-09-22 05:40:51.000000000 +0200
+++ new/gtk+-3.14.0/util/Makefile.in	2014-10-06 16:44:31.524509760 +0200
@@ -195,18 +195,18 @@
 CAIRO_REQUIRED_VERSION = @CAIRO_REQUIRED_VERSION@
 CATALOGS = @CATALOGS@
 CATOBJEXT = @CATOBJEXT@
-CC = @CC@
+CC = @CC_FOR_BUILD@
 CCAS = @CCAS@
 CCASDEPMODE = @CCASDEPMODE@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS_FOR_BUILD@
 CFLAGS_FOR_BUILD = @CFLAGS_FOR_BUILD@
 COLORD_CFLAGS = @COLORD_CFLAGS@
 COLORD_LIBS = @COLORD_LIBS@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS_FOR_BUILD@
 CPPFLAGS_FOR_BUILD = @CPPFLAGS_FOR_BUILD@
 CPP_FOR_BUILD = @CPP_FOR_BUILD@
 CUPS_API_MAJOR = @CUPS_API_MAJOR@
@@ -313,7 +313,7 @@
 JSON_GLIB_CFLAGS = @JSON_GLIB_CFLAGS@
 JSON_GLIB_LIBS = @JSON_GLIB_LIBS@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS_FOR_BUILD@
 LDFLAGS_FOR_BUILD = @LDFLAGS_FOR_BUILD@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
@@ -485,7 +485,7 @@
 
 extract-strings$(EXEEXT): $(extract_strings_OBJECTS) $(extract_strings_DEPENDENCIES) $(EXTRA_extract_strings_DEPENDENCIES) 
 	@rm -f extract-strings$(EXEEXT)
-	$(AM_V_CCLD)$(extract_strings_LINK) $(extract_strings_OBJECTS) $(extract_strings_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(extract_strings_LINK) $(extract_strings_OBJECTS) $(extract_strings_LDADD)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
