diff -Naur old/openssl-1.0.1h/crypto/ui/ui_openssl.c new/openssl-1.0.1h/crypto/ui/ui_openssl.c
--- old/openssl-1.0.1h/crypto/ui/ui_openssl.c	2014-06-05 11:44:33.000000000 +0200
+++ new/openssl-1.0.1h/crypto/ui/ui_openssl.c	2014-06-21 22:36:55.208725066 +0200
@@ -220,6 +220,9 @@
 #undef SGTTY
 #endif
 
+#undef TERMIO
+#define TERMIOS
+
 #ifdef TERMIOS
 # include <termios.h>
 # define TTY_STRUCT		struct termios
