--- old/toybox-0.4.9/toys/lsb/killall.c	2014-07-07 06:43:29.000000000 +0200
+++ new/toybox-0.4.9/toys/lsb/killall.c	2014-09-20 23:35:45.607276762 +0200
@@ -72,7 +72,7 @@
     return;
   }
 
-  if (TT.sig || **TT.names == '-') {
+  if (TT.sig || (*TT.names && **TT.names == '-')) {
     if (0 > (TT.signum = sig_to_num(TT.sig ? TT.sig : (*TT.names)+1))) {
       if (toys.optflags & FLAG_q) exit(1);
       error_exit("Invalid signal");
