diff -Naur old/uClibc++-0.2.4/Rules.mak new/uClibc++-0.2.4/Rules.mak
--- old/uClibc++-0.2.4/Rules.mak	2012-05-25 23:15:31.000000000 +0200
+++ new/uClibc++-0.2.4/Rules.mak	2014-06-27 22:08:21.809222126 +0200
@@ -80,7 +80,7 @@
         then echo "-DGCC_HASCLASSVISIBILITY"; fi)
 
 check_as_needed=$(shell if $(LD) --help | grep -q 'as-needed' ; \
-	then echo "-Wl,--as-needed -lgcc_s -Wl,--no-as-needed"; else echo "-lgcc_s"; fi)
+	then echo "-Wl,--as-needed -lgcc_eh -Wl,--no-as-needed"; else echo "-lgcc_eh"; fi)
 
 # strip quotes
 qstrip = $(strip $(subst ",,$(1)))
